crates/polars-io/src/ipc/ipc_file.rs

```diff
@@ -164,10 +164,10 @@ impl<R: MmapBytesReader> IpcReader<R> {
         let rechunk = self.rechunk;
         let metadata = read::read_file_metadata(&mut self.reader)?;
 
-        debug_assert!(
-            self.columns.is_none(),
-            "column names must have already been converted into indices",
-        );
+        // TODO: Replace this back with the assertion before merging https://github.com/pola-rs/polars/pull/14861.
+        if let Some(columns) = &self.columns {
+            self.projection = Some(columns_to_projection(columns, &metadata.schema)?);
+        }
```

Comment:
Reusing `finish_with_scan_ops` for the simple implementation. Won't need in the final version.

Reply:
Do you think we should keep this change? Instead of asserting that the conversion already occurred, to implement the conversion again? This means that for some code paths, the conversion code is hit twice (which is not a functional problem since it only gets executed once, but more of a software quality issue).

Reply:
If you can keep the `assert!` that's fine. I don't know why the change is introduced then?

Reply:
Because 

> Reusing `finish_with_scan_ops` for the simple implementation.

The current implementation uses the sync IpcReader internally to parse the binary data and the predicate option requires using `finish_with_scan_ops` but `finish_with_scan_ops` requires the conversion from strings to integers to have been done.